src/applications/disability-benefits/526EZ/reducer.js

```diff
@@ -1,6 +1,58 @@
+
+import _ from 'lodash/fp';
+
 import formConfig from './config/form';
 import { createSaveInProgressFormReducer } from '../../common/schemaform/save-in-progress/reducers';
 
+import {
+  SET_PRESTART_STATUS,
+  RESET_PRESTART_STATUS,
+  RESET_PRESTART_DISPLAY,
+  PRESTART_STATUSES
+} from './actions';
+
+const initialState = {
+  status: PRESTART_STATUSES.notAttempted,
+  data: undefined,
+  display: false
+};
+
+export const prestart = (state = initialState, action) => {
+  switch (action.type) {
+    case SET_PRESTART_STATUS: {
+      const newState = _.set('status', action.status, state);
+      if (!state.display) {
+        newState.display = true;
```

Comment:
This will always set `newState.display` to truthy; either it's truthy to begin with or it's true now.

If there's no reason to pass a falsey `display` value, can we remove the check altogether?
If this is to set a default value, you might be looking for `if (state.display === undefined)`.

Reply:
Because we've got subsequent requests being made (`submit` after `check`), it's not necessary to set display to true once it's already been set by an earlier call.

Reply:
I think checking the old state will take as much time as always setting the new state, so I'm not sure there is a whole lot to be gained here. Conversely, I think this will make it a bit harder to reason about. It sounds like we always want the `newState.display` to be true, so we should always set it to `true`.

I think @kreek's allergy to conditionals is contagious. I think I've caught it.

Reply:
join [anti-if](https://francescocirillo.com/pages/anti-if-campaign) ✊- in this case _if_ it always  can be set to true without any adverse effects I would lose the conditional check just to reduce lines a follow developer needs to process

Reply:
Sounds good, will update, thanks for sharing the link and the insights.